pkgs/os-specific/linux/lvm2/default.nix

```diff
@@ -24,7 +24,8 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ udev libuuid thin-provisioning-tools libaio ];
+  buildInputs = [ libuuid thin-provisioning-tools libaio ]
+                ++ lib.optional (!stdenv.hostPlatform.isMusl) [ udev ];
```

Comment:
```
checking whether gcc accepts -Wjump-misses-init flag... yes                                                                                                         [20/45514]checking whether gcc accepts -Wclobbered flag... yes                                                                                                                          checking whether gcc accepts -Wsync-nand flag... yes                                                                                                                          checking for C optimisation flag... -O2                                                                                                                                       checking whether to gather gcov profiling data... no                                                                                                                          checking whether to enable valgrind awareness of pools... no                                                                                                                  checking pkg-config is at least version 0.9.0... yes                                                                                                                          checking for PKGCONFIGINIT... no                                                                                                                                              pkg-config initialized                                                                                                                                                        checking for VALGRIND... no                                                                                                                                                   checking whether to use device-mapper... yes                                                                                                                                  checking whether to build lvmpolld... no                                                                                                                                      checking whether to build lvmlockdsanlock... no                                                                                                                               checking whether to build lvmlockddlm... no                                                                                                                                   checking whether to build lvmlockddlmcontrol... no                                                                                                                            checking whether to build lvmlockd... no                                                                                                                                      checking whether to build dmfilemapd... no                                                                                                                                    checking whether to build notifydbus... no                                                                                                                                    checking for BLKID... yes                                                                                                                                                     checking whether to enable libblkid detection of signatures when wiping... yes                                                                                                checking for SYSTEMD... no                                                                                                                                                    checking whether to use udev-systemd protocol for jobs in background... no
checking whether to enable synchronisation with udev processing... yes
checking for UDEV... no
configure: error: Package requirements (libudev >= 143) were not met:

No package 'libudev' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables UDEV_CFLAGS
and UDEV_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

Reply:
fixed.